### PR TITLE
Add docker compose user guide [skip ci]

### PR DESCRIPTION
**Pull request by IsaacYangSLA**

### CODE REVIEW

File: docs/user_guide/docker_compose.rst [hunk 1]
```diff
@@ -0,0 +1,184 @@
+.. _docker_compose:
+
+######################################################
+Launching NVIDIA FLARE with docker compose
+######################################################
+
+For users who would like to run NVIDIA FLARE locally and already have docker 
+environment ready, the provisioning tool of NVIDIA FLARE now includes a new
+builder ``DockerBuilder`` that can create ``compose.yaml`` and other information.  
+After provisioing, users can enter the result folder, normally in
```
[the remark this answers is not in the document]
Comment (kkersten) [reply docs/user_guide/docker_compose.rst:10]: I agree that we should describe the use case, but is this accurate?  I don't think the kits used in this case could be shipped out.  Wouldn't this require re-provisioning for a true distributed deployment?

[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply docs/user_guide/docker_compose.rst:10]: The only use case I could think of is the convenience to users.  Just one machine, two commands, and the system is up and running.  With the admin console, even it's on a different machine, the user can start running almost all examples.  Basically, it removes lots of steps during 'getting started' stage.

[the remark this answers is not in the document]
Comment (kkersten) [reply docs/user_guide/docker_compose.rst:10]: Agree, I think that will be the primary use case.  Test with a secure, local deployment.  Looks good.

**Review by nvkevlu — APPROVED** [review]
Thank you, LGTM

**Review by nvkevlu — APPROVED** [review]

**Pull request merged**